core/src/main/scala-2/chisel3/Contract.scala

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.SourceInfo
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl.ir._
+import scala.language.experimental.macros
+import scala.reflect.macros.whitebox
+
+object Contract {
```

Comment:
can this be called `FormalContract`? A contract is a very abstract term, and we also had the idea of chisel `Interface` providing a different kind of contract...

Reply:
Yes definitely, I can changed that to `FormalContract` :+1: